includes/wikia/parser/templatetypes/handlers/InfoIconTemplate.class.php

```diff
@@ -20,16 +20,34 @@ public static function handle( $wikitext, Parser $parser ) {
 		
 		$output = '';
 		foreach ( $images as $image ) {
-			$title = Title::newFromText( $image );
-			$file = wfFindFile( $title );
-			$output .= Linker::makeImageLink2(
-				$title,
-				$file,
-				[],
-				[ 'template-type' => AutomaticTemplateTypes::TEMPLATE_INFOICON ]
-			);
+			$output .= self::makeIconLink( $image );
 		}
 		$stripMarker = $parser->insertStripItem( $output );
 		return $stripMarker;
 	}
+
+	/**
+	 * @desc for a given wikitext image mark, create imagelink
+	 *
+	 * @param $image
+	 * @return String
+	 * @throws \MWException
+	 */
+	private static function makeIconLink( $image ) {
```

Comment:
only idea which comes to my mind is to use `return` twice here, but it's even worse -.-